hazelcast/src/main/java/com/hazelcast/config/ConfigBuilder.java

```diff
@@ -19,7 +19,7 @@
 /**
  * Interface for all config builders.
  */
-
+@FunctionalInterface
 public interface ConfigBuilder {
```

Comment:
Why do we need to mark interfaces for which we provide the implementations (like this one, or `ICacheManager`, `LocalInstanceStats`, `AuthenticationConfig` (I think), etc.) as functional? And why would we need to protect them from intorducing new methods?

Reply:
You're right, we don't need to mark those. I'll remove the annotations.

Reply:
I've removed the annotation for multiple interfaces in a separate commit, please take a look. I may have mistakenly kept the annotation for one or two interfaces that I'm unsure if it's meant to be implemented by the user but better that we mark an internal interface by mistake than not mark a public interface.